### PR TITLE
Add a resource list section component

### PR DESCRIPTION
**Pull request by matvlaovi-flip**
Added a resource-list-section component.
Drag handles won't be display for items inside a section so dragging won't be possible.

### CODE REVIEW

File: packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  display: block;
+  width: 100%;
+
+  & * {
+    box-sizing: border-box;
+  }
+}
+
+.resource-list-section {
+  background-color: var(--s-surface-overlay-default);
```
[the remark this answers is not in the document]
Comment (matvlaovi-flip) [reply packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css:13]: Good questions, I'll check with Liu. Went by the action-item-list but that one doesn't have the spacing so this is a bit of a different use case

File: packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+.resource-list-section__label {
```
[the remark this answers is not in the document]
Comment (matvlaovi-flip) [reply packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css:16]: Will ask Liu

File: packages/swirl-components/src/components/swirl-resource-list/swirl-resource-list.tsx [hunk 3]
```diff
@@ -54,6 +55,7 @@ export class SwirlResourceList {
     this.setupControllingElement();
     this.setItemAllowDragState();
     this.setupDragDrop();
+    this.propagateSpacingToSections();
```
[the remark this answers is not in the document]
Comment (matvlaovi-flip) [reply packages/swirl-components/src/components/swirl-resource-list/swirl-resource-list.tsx:58]: Yup makes sense

[the remark this answers is not in the document]
Comment (matvlaovi-flip) [reply packages/swirl-components/src/components/swirl-resource-list/swirl-resource-list.tsx]: done

**Comment by matvlaovi-flip** [comment]
@Sqrrl there were some design changes https://www.figma.com/design/JC7QsUEZukSDZytvRjrqIh/UX-Improvements?node-id=86-12557&node-type=canvas&m=dev

Section label will no longer have a bottom border, but a separator above it, if it's not the first one in the list. Added a prop to the section for the separator, and the parent list will manage whether or not it will have a separator

File: packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (matvlaovi-flip) [reply packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css]: Design changed, there will now be a swirl-separator above

File: packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (matvlaovi-flip) [reply packages/swirl-components/src/components/swirl-resource-list-section/swirl-resource-list-section.css]: design changed and component adjusted

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**